Common/src/gRPC/Services/Agent.cs

```diff
@@ -563,6 +565,41 @@ await completionTask.WaitAsync(cancellationToken)
     return new ResultReply();
   }
 
+  /// <inheritdoc />
+  public async Task<CreateResultsMetaDataResponse> CreateResultsMetaData(CreateResultsMetaDataRequest request,
+                                                                         CancellationToken            cancellationToken)
+  {
+    var results = request.Results.Select(rc => new Storage.Result(request.SessionId,
+                                                                  Guid.NewGuid()
+                                                                      .ToString(),
+                                                                  rc.Name,
+                                                                  "",
+                                                                  ResultStatus.Created,
+                                                                  new List<string>(),
+                                                                  DateTime.UtcNow,
+                                                                  Array.Empty<byte>()))
+                         .ToList();
```

Comment:
Do you need the ToList here? I think resultTable.Create takes an IEnumerable, so you could give the Select as is.

Reply:
Yes! I'm iterating two times over the results here; first during insertion in DB, second during return. Without the `ToList`, the guid would be generated twice and the result created and the results returned would not be the same.

Reply:
Ah. Good call.